src/Files.App/Data/Interacts/BaseLayoutCommandsViewModel.cs

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+using Files.App.Filesystem.StorageItems;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Input;
+using System.IO;
+using System.Windows.Input;
+using Windows.ApplicationModel.DataTransfer;
+using Windows.ApplicationModel.DataTransfer.DragDrop;
+using Windows.Storage;
+using Windows.System;
+using static Files.App.Constants.Browser.GridViewBrowser;
+
+namespace Files.App.Data.Interacts
+{
+	public class BaseLayoutCommandsViewModel : IDisposable
+	{
+		private readonly IShellPage _associatedInstance;
+
+		private readonly ItemManipulationModel _itemManipulationModel;
+
+		public ICommand CreateNewFileCommand { get; private set; }
+
+		public ICommand ItemPointerPressedCommand { get; private set; }
+
+		public ICommand PointerWheelChangedCommand { get; private set; }
+
+		public ICommand DragOverCommand { get; private set; }
+
+		public ICommand DropCommand { get; private set; }
+
+		public BaseLayoutCommandsViewModel(IShellPage associatedInstance, ItemManipulationModel itemManipulationModel)
+		{
+			_associatedInstance = associatedInstance;
+			_itemManipulationModel = itemManipulationModel;
+
+			InitializeCommands();
+		}
+
+		private void InitializeCommands()
+		{
+			CreateNewFileCommand = new RelayCommand<ShellNewEntry>(CreateNewFile);
+			ItemPointerPressedCommand = new AsyncRelayCommand<PointerRoutedEventArgs>(ItemPointerPressed);
+			PointerWheelChangedCommand = new RelayCommand<PointerRoutedEventArgs>(PointerWheelChanged);
+			DragOverCommand = new AsyncRelayCommand<DragEventArgs>(DragOver);
+			DropCommand = new AsyncRelayCommand<DragEventArgs>(Drop);
+		}
+
+		private void CreateNewFile(ShellNewEntry f)
+		{
+			UIFilesystemHelpers.CreateFileFromDialogResultType(AddItemDialogItemType.File, f, _associatedInstance);
+		}
+
+		private async Task ItemPointerPressed(PointerRoutedEventArgs e)
+		{
+			if (e.GetCurrentPoint(null).Properties.IsMiddleButtonPressed)
```

Comment:
Could this be inverted to reduce nesting?

Reply:
Fixed